odoo/fields.py

```diff
@@ -2950,6 +2950,13 @@ def setup_nonrelated(self, model):
             _logger.warning("Field %s with unknown comodel_name %r", self, self.comodel_name)
             self.comodel_name = '_unknown'
 
+    def setup(self, model):
+        super().setup(model)
+        if self.check_company and 'company_id' not in model._fields and model._name != 'res.company' and self.comodel_name != 'res.users' and not self.company_dependent:
```

Comment:
Wouldn't there be a way to check that `_check_company_domain` is valid in order to avoid having to know about the model names in `fields.py`?

Reply:
I don't know. My code works.

Reply:
But `fields.py` shouldn't need to know about model names.

Reply:
And `res.lang` is used here:
https://github.com/odoo/odoo/blob/9781e148d7187dd45961a52c9e2fdbc48985c174/odoo/fields.py#L1869

I think both `fields.py` and `models.py` should be allowed to use `base` module model names.

See `res.users` here:
https://github.com/odoo/odoo/blob/9781e148d7187dd45961a52c9e2fdbc48985c174/odoo/models.py#L252